src/OpenTok/Util/Plugin/OpentokAuth.php

```diff
@@ -34,14 +34,19 @@ public function onBeforeSend(Event $event)
 
     private function createAuthHeader()
     {
-      $token = array(
-          'ist' => 'project',
-          'iss' => $this->apiKey,
-          'iat' => time(),
-          'exp' => time()+(5 * 60),
-          'jti' => (float) rand() / (float) getrandmax(),
-      );
-
-      return JWT::encode($token, $this->apiSecret);
+        $token = array(
+            'ist' => 'project',
+            'iss' => $this->apiKey,
+            'iat' => time(), // this is in seconds
+            'exp' => time()+(5 * 60),
+            'jti' => $this->generateNonce(),
+        );
+        return JWT::encode($token, $this->apiSecret);
+    }
+
+    private function generateNonce($len = 10) {
+        $bytes = openssl_random_pseudo_bytes($len);
```

Comment:
the nonce doesn't need to come from a cryptographically secure (significant entropy) psuedo-random number generator. this is unnecessarily slow. FYI: if you are going to depend on having openssl on the system, its best to [declare this as a dependency](https://getcomposer.org/doc/02-libraries.md#platform-packages) in the `composer.json`.

its actually recommended to use a time-dependent random string, so there is zero chance of a repeating value. the whole purpose of a nonce is not for randomness, but for non-repetition. this is precisely why i linked to the http://php.net/manual/en/function.uniqid.php function. it's also faster.

sorry, i didn't mean to be nitpicky 😛